### PR TITLE
Modularize and add TS file for LDrawLoader

### PR DESCRIPTION
**Pull request by yomboprime**
Also added @gkjohnson as LDrawLoader author.

### CODE REVIEW

File: examples/jsm/loaders/LDrawLoader.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+} from '../../../src/Three';
+
+export class LDrawLoader {
+	constructor(manager?: LoadingManager);
+	manager: LoadingManager;
+	path: string;
+
+	load(url: string, onLoad: (data: Group) => void, onProgress?: (event: ProgressEvent) => void, onError?: (event: ErrorEvent) => void) : void;
+	setPath(path: string) : this;
+	setFileMap(fileMap: Record<string, string>): void;
```
Comment (yomboprime) [inline examples/jsm/loaders/LDrawLoader.d.ts]: Is it ok to use `Record` to express a hash object here? (string -> string)

Comment (Mugen87) [reply examples/jsm/loaders/LDrawLoader.d.ts]: I've not used `Record` so far but I think it's appropriate for this context.

File: examples/jsm/loaders/LDrawLoader.d.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	setMaterials(materials: Material[]): void;
+
+	parse(text: string, path?: string, onLoad: (data: Group) => void): void;
```
Comment (Mugen87) [inline examples/jsm/loaders/LDrawLoader.d.ts]: The required parameter `onLoad` can not follow the optional parameter (`path`).

**Comment by Mugen87** [comment]
It seems you can put `Cache`, `Material` and `Object3D` on the ignore list.

File: examples/jsm/loaders/LDrawLoader.d.ts (hunk 2 above)
Comment (yomboprime) [reply examples/jsm/loaders/LDrawLoader.d.ts]: Done.

**Comment by yomboprime** [comment]
> It seems you can put `Cache`, `Material` and `Object3D` on the ignore list.

Done. Thanks!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**